pkg/webui/console/components/payload-formatters-form/index.js

```diff
@@ -197,13 +204,46 @@ class PayloadFormattersForm extends React.Component {
     }
   }
 
+  @bind
+  pastePayloadFormatter(app) {
+    const { defaultParameter, uplink, repoFormatters } = this.props
+    const applicationFormatter = defaultParameter
+      ? defaultParameter
+      : getDefaultJavascriptFormatter(uplink)
+    if (app && this.formRef !== null) {
+      return () =>
+        this.formRef?.current?.setFieldValue(FIELD_NAMES.JAVASCRIPT, applicationFormatter)
+    }
+
+    return () =>
+      this.formRef?.current?.setFieldValue(
+        FIELD_NAMES.JAVASCRIPT,
+        repoFormatters.formatter_parameter,
+      )
+  }
```

Comment:
Mind the complexity/readability/maintainability costs of making unified handlers vs explicitly writing distinct handlers:

```javascript
  @bind
  pasteAppPayloadFormatter() {
    const { defaultParameter, uplink } = this.props
    this.formRef?.current?.setFieldValue(FIELD_NAMES.JAVASCRIPT, defaultParameter || getDefaultJavascriptFormatter(uplink))
  }

  @bind
  pasteRepoPayloadFormatter() {
    const { repoFormatters } = this.props
    this.formRef?.current?.setFieldValue(
      FIELD_NAMES.JAVASCRIPT,
      repoFormatters.formatter_parameter,
    )
  }
```

Don't sacrifice code clarity for a tiny memory improvement (if any).